src/pangi/application/services/model_routing.py

```
@@ -376,13 +383,48 @@ def _block(
+            raise ModelProviderFailure(
+                code,
+                retryable=False,
+                provider_request_count=response.provider_request_count,
+                duration_ms=response.duration_ms,
+            )
+        try:
+            valid = self._output_validator.is_valid(
+                schema=guarded.output_schema,
+                canonical_output_json=response.canonical_output_json,
+            )
+        except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)